src/app/dog/dog.component.ts

```diff
@@ -0,0 +1,46 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute, Params, Router } from '@angular/router';
+import { DogsService } from '../dogs/dogs.service';
```

Comment:
leave an empty line between imports from angular or 3rds parties and your own components